1/src/components/counter/conteiner/CounterConteiner.js

```diff
@@ -0,0 +1,57 @@
+import {Component} from 'react';
+import { CounterView } from '../component/CounterView';
+
+class CounterContainer extends Component {
+    constructor(props) {
+        super(props)
+
+
+        this.state = {
+            countValue : 0,
+            isEven : true
+        }
+    }
+
+    componentDidUpdate() {
+        if(this.state.countValue % 2 === 0 && this.state.isEven !== true) {
+            this.setState({isEven : true})
+        } 
+        if (this.state.countValue % 2 !== 0 && this.state.isEven !== false) {
+            this.setState({isEven : false})
+        }
+
+    }
+
+    handleIncrement = () => {   
+        this.setState({countValue : this.state.countValue + 1})
+    }
+
+    handleReset = () => {   
+        this.setState({countValue : 0})
+    }
+
+    handleSubtraction = () => {   
+        this.setState((prevstate) => {
+            if (prevstate.countValue - 1 < 0) {
+                return {countValue : 0 }
+            } else {
+                return {countValue : prevstate.countValue - 1}
+            }
+        })
```

Comment:
rework you method to optimize the following: Trigger setState only if <statement>. In your case you call setState every single time. That's not good